src/replace-positional-args.js

```diff
@@ -0,0 +1,36 @@
+// Loops over script tokens and replace any position args where positional args are denoted by $<number> and are 1
+// indexed (like bash scripts), eg `echo hello $1 world`
+// eslint-disable-next-line complexity
+function replacePositionalArgs(script, args) {
+  if (!script) {
+    return [script, args]
+  }
+  const scriptTokens = script.split(' ')
+  const usedArgs = []
+  for (const [i, token] of scriptTokens.entries()) {
+    if (token[0] !== '$') {
+      // eslint-disable-next-line no-continue
+      continue
+    }
+    const argIndex = parseInt(token.slice(1), 10)
+    const arg = args[argIndex]
+    if (!arg) {
```

Comment:
What if arg is `false` or an empty string? Do we want to do a typeof undefined check instead?

Reply:
good point will fix